views/components/ConfigurationsList/index.jsx

```diff
@@ -0,0 +1,840 @@
+        return { ...prevState, gridCols: 1 }
+      })
+    }
+  }
+  const babyProofPreferences = (preferences) => {
+    let preference = {}
+    preference['star'] = 'star' in preferences ? preferences['star'] : {}
+    preference['sort'] = 'sort' in preferences ? preferences['sort'] : 0
+    preference['config'] = 'config' in preferences ? preferences['config'] : ''
+    preference['complete'] =
+      'complete' in preferences ? preferences['complete'] : {}
+    return preference
+  }
+  const updateUserPreferences = (index, type) => {
+    let uid = user.uid
+    let preference = {}
+    if (type == 'index') {
+      if (state.configurations.length > 0 && state.configurations[index]) {
+        preference['config'] = state.configurations[index]['_id']
+      }
+    } else {
+      preference['config'] = index
+    }
+    preference['complete'] =
+      'complete' in state.preferences ? state.preferences['complete'] : {}
+    preference['star'] =
+      'star' in state.preferences ? state.preferences['star'] : {}
+    preference['sort'] =
+      'sort' in state.preferences ? state.preferences['sort'] : 0
+    preference = babyProofPreferences(preference)
+
+    return window
+      .fetch(`${basePath}/api/v1/users/${uid}/preferences`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'same-origin',
+        body: JSON.stringify({
+          preferences: preference,
+        }),
+      })
+      .then(() => {
+        if (type == 'index') {
+          setState((prevState) => {
+            return {
+              ...prevState,
+              preferences: preference,
+              snackTime: true,
+            }
+          })
+        } else {
+          setState((prevState) => {
+            return {
+              ...prevState,
+              preferences: preference,
+            }
+          })
+        }
+      })
+  }
+  const fetchPreferences = (uid) => {
+    return window
+      .fetch(`${basePath}/api/v1/users/${uid}/preferences`, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'same-origin',
+      })
+      .then((response) => {
+        if (response.status !== 200) {
+          return
+        }
+        return response.json()
+      })
+      .then((response) => {
+        setState((prevState) => {
+          return {
+            ...prevState,
+            preferences: babyProofPreferences(response),
+          }
+        })
+      })
+  }
+  const fetchConfigItem = (uid, _id) => {
+    return window
+      .fetch(`${basePath}/api/v1/users/${uid}/configs/${_id}`, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'same-origin',
+      })
+      .then((response) => {
+        if (response.status !== 200) {
+          return
+        }
+        return response.json()
+      })
+      .then(() => {})
```

Comment:
What does the extra `then` do?

Reply:
good find thanks [ee04cc1](https://github.com/AMP-SCZ/dpdash/pull/304/commits/ee04cc1f53156889ea38ceef5de6725ff205f44f)